### PR TITLE
ARXIVNG-2622 ARXIVNG-2623 adds integration module for preview service

### LINKED ISSUE #72: Add service integration module for preview service
The submission preview service is getting close to a minimum viable version: https://github.com/arXiv/arxiv-submission-preview . We require a service integration module for reading/depositing previews, using its API.

### PR DESCRIPTION
**Pull request by erickpeirson**
Fixes #72 . This currently supports the [``getServiceStatus``](https://github.com/arXiv/arxiv-submission-preview/blob/fc5c5d53e7655be4abbc138fa589a0dc85f20222/schema/openapi.yaml#L18),  [``depositSubmissionPreview``](https://github.com/arXiv/arxiv-submission-preview/blob/fc5c5d53e7655be4abbc138fa589a0dc85f20222/schema/openapi.yaml#L62) and [``getSubmissionPreview``](https://github.com/arXiv/arxiv-submission-preview/blob/fc5c5d53e7655be4abbc138fa589a0dc85f20222/schema/openapi.yaml#L48) operations.

### CODE REVIEW

**Comment by erickpeirson** [comment]
This is implemented against https://github.com/arXiv/arxiv-submission-preview/pull/21 (https://github.com/arXiv/arxiv-submission-preview/pull/21/commits/2403e9d6dc5f5d13c46d507a9699e9efee3e2e5c).

File: core/arxiv/submission/domain/compilation.py [hunk 1]
```diff
@@ -93,6 +93,26 @@ def content_type(self):
         """Get the MIME type for the compilation product."""
         return self.output_format.content_type
 
+    @property
+    def is_succeeded(self) -> bool:
```
Comment (mhl10) [inline core/arxiv/submission/domain/compilation.py:101]: Should this domain class throw an error if there are obvious logic problems? e.g. `is_succeeded` and `is_failed` can't be true at the same time, and `is_finished` and `is_in_progress` can't be true at the same time.

**Review by mhl10 — APPROVED** [review]
Looks good! One minor comment.

File: core/arxiv/submission/domain/compilation.py (hunk 1 above)
Comment (erickpeirson) [reply core/arxiv/submission/domain/compilation.py:101]: Fixed in d9c2ff3

**Pull request merged**